synthetic_data/distinct_generators/datetime_generator.py

```diff
@@ -32,13 +34,14 @@ def generate_datetime(
     """
     t = rng.random()
     ptime = start_date + t * (end_date - start_date)
+    ptime = ptime.strftime(date_format)
 
-    return ptime.strftime(date_format)
+    return [date_format, datetime.strptime(ptime, date_format)]
 
```

Comment:
this will return arrays in this format: [date_format, datetime_object]

This is to allow for sorting and converting back to string. (there's no way to know what format an already made datetime object is, so each item in datetime array returned by `random_datetimes` will save both the format and datetime object.

Reply:
oh, there is an issue with how to return it when you don't end up sorting it... it will remain in this unusable form...
Possibly alter it's form during the loop in `generate_dataset_by_class`?